cprover_bindings/src/goto_program/typ.rs

```diff
@@ -539,6 +568,13 @@ impl Type {
         }
     }
 
+    pub fn is_union_like(&self) -> bool {
+        match self {
```

Comment:
nit: This could be converted into:

```rust
matches!(self, IncompleteUnion { .. } | Union { .. } | UnionTag(_))
```

Reply:
My concern with that pattern is that the compiler wouldn't warn if we add a new variant to the `Type` enum